src/contexts/Drive/Drive.context.tsx

```diff
@@ -46,6 +48,52 @@ interface DriveContextProviderProps {
 const logger = new BaseLogger({
   tag: 'DRIVE_CONTEXT',
 });
+
+export const removeOldItemsFromLocalDB = async () => {
+  const lastUpdatedAt = (await asyncStorageService.getItem(AsyncStorageKey.LastUpdatedAt)) || new Date().toISOString();
+
+  const [modifiedFiles, modifiedFolders] = await Promise.all([
+    driveFileService.getModifiedFiles({
+      updatedAt: lastUpdatedAt,
+      status: 'ALL',
+    }),
+    driveFolderService.getModifiedFolders({
+      updatedAt: lastUpdatedAt,
+      status: 'ALL',
+    }),
+  ]);
+
+  if (!modifiedFiles || !modifiedFolders) return;
+
+  const modifiedFilesIds = modifiedFiles.map((file) => file.id);
+  const modifiedFoldersIds = modifiedFolders.map((folder) => folder.id);
+
+  modifiedFilesIds.forEach((fileId) => {
+    driveLocalDB.deleteItem({ id: fileId }).catch((error) => {
+      errorService.reportError(error);
+    });
+  });
+
+  modifiedFoldersIds.forEach((folderId) => {
+    driveLocalDB.deleteFolderRecord(folderId).catch((error) => {
```

Comment:
Don't do this, instead wait for the operation to complete blocking the code execution here's why:

```
// This timeout represents an async funcion, that is not being awaited
setTimeout(() => {
   deleteFolderRecord()
}, 500)

// If we read and the delete has not finished, we will still get the not yet deleted item
getDriveLocalDBItems()